server/services/source-map.js

```diff
@@ -1,28 +1,70 @@
 import Log from './../context/log';
 import Conf from './../context/conf';
 import fs from 'fs';
+import path from 'path';
 
 const istanbulAPI = Npm.require('istanbul-api');
 const libSourceMaps = istanbulAPI.libSourceMaps;
 
 const sourceMap = libSourceMaps.createSourceMapStore({verbose: Conf.IS_COVERAGE_ACTIVE});
 const meteor_dir = Conf.COVERAGE_APP_FOLDER;
-const regexAlterationSourceMapPath = new RegExp(/(packages\/)([a-zA-Z-]*)[_:]([a-zA-Z-_]*)(.*)/);
+// Next RegExps must consider following paths:
+//  1. meteor://💻app/packages/lmieulet:meteor-coverage/server/index.js
+//  2. meteor://💻app/.npm/package/node_modules/minimatch/package.json
+//  3. meteor://💻app/node_modules/meteor/lmieulet:meteor-coverage/node_modules/minimatch/minimatch.js
+//  4. meteor://💻app/packages/local-test:lmieulet:meteor-coverage/server/tests.js
+//  5. meteor://💻app/node_modules/meteor/local-test:cgalvarez:my-package/tests/client/mocks.js
+//  6. meteor://💻app/node_modules/meteor/local-test:cgalvarez:my-package/node_modules/chai-as-promised/lib/chai-as-promised.js
+const regexAlterationSourceMapPath = new RegExp(/^(packages\/|node_modules\/meteor\/)(?:local-test[_:])?([a-zA-Z-]*)[_:]([a-zA-Z-_]*)(.*)$/);
+const regexPUT = new RegExp(/packages\/local-test:([a-zA-Z-_]*):([a-zA-Z-_]*)\/.*/);
 
 // Alter inside the source map the path of each sources
 alterSourceMapPaths = function (map) {
   var match;
+  if (!this.PUT && Meteor.isPackageTest && meteor_dir === `${process.env.PWD}/`) {
```

Comment:
This `if` must be `if (!this.PUT && Meteor.isPackageTest && meteor_dir.replace(/\/$/, '') === process.env.PWD) {`.

When executing the script `test:packages-coverage-watch`, `meteor_dir` does not contain a trailing slash, which casuses to not detect the PUT. To cover both cases, just use a regex to remove the trailing slash when present and compare it to `process.env.PWD` directly.